src/System.EventLog/src/System/Diagnostics/EventLogInternal.cs

```diff
@@ -0,0 +1,1550 @@
+        {
+            if (dllNameList == null)
+                return null;
+
+            if (insertionStrings == null)
+                insertionStrings = new string[0];
+
+            string[] listDll = dllNameList.Split(';');
+
+            // Find first mesage in DLL list
+            foreach (string dllName in listDll)
+            {
+                if (dllName == null || dllName.Length == 0)
+                    continue;
+
+                SafeLibraryHandle hModule = null;
+
+                if (IsOpen)
+                {
+                    hModule = MessageLibraries[dllName] as SafeLibraryHandle;
+
+                    if (hModule == null || hModule.IsInvalid)
+                    {
+                        hModule = Interop.Kernel32.LoadLibraryExW(dllName, IntPtr.Zero, NativeMethods.LOAD_LIBRARY_AS_DATAFILE);
+                        MessageLibraries[dllName] = hModule;
+                    }
+                }
+                else
+                {
+                    hModule = Interop.Kernel32.LoadLibraryExW(dllName, IntPtr.Zero, NativeMethods.LOAD_LIBRARY_AS_DATAFILE);
+                }
+
+                if (hModule.IsInvalid)
+                    continue;
+
+                string msg = null;
+                try
+                {
+                    msg = EventLog.TryFormatMessage(hModule, messageNum, insertionStrings);
+                }
+                finally
+                {
+                    if (!IsOpen)
+                    {
+                        hModule.Close();
+                    }
+                }
+
+                if (msg != null)
+                {
+                    return msg;
+                }
+
+            }
+            return null;
+        }
+
+        internal EventLogEntry[] GetAllEntries()
+        {
+            // we could just call getEntryAt() on all the entries, but it'll be faster
+            // if we grab multiple entries at once.
+            string currentMachineName = this.machineName;
+
+            if (!IsOpenForRead)
+                OpenForRead(currentMachineName);
+
+            EventLogEntry[] entries = new EventLogEntry[EntryCount];
+            int idx = 0;
+            int oldestEntry = OldestEntryNumber;
+
+            int bytesRead;
+            int minBytesNeeded;
+            int error = 0;
+            while (idx < entries.Length)
+            {
+                byte[] buf = new byte[BUF_SIZE];
+                bool success = UnsafeNativeMethods.ReadEventLog(readHandle, NativeMethods.FORWARDS_READ | NativeMethods.SEEK_READ,
+                                                      oldestEntry + idx, buf, buf.Length, out bytesRead, out minBytesNeeded);
+                if (!success)
+                {
+                    error = Marshal.GetLastWin32Error();
+                    Debug.WriteLineIf(CompModSwitches.EventLog.TraceVerbose, "Error from ReadEventLog is " + error.ToString(CultureInfo.InvariantCulture));
+#if !RETRY_ON_ALL_ERRORS
```

Comment:
RETRY_ON_ALL_ERRORS is never defined, so please remove the #if block, leaving the contents.

Reply:
Removed withy contents remaining